src/imp/schannel.rs

```diff
@@ -559,4 +688,119 @@ mod pem {
             vec![b"-----BEGIN garbage" as &[u8]]
         );
     }
-}
+
+    #[test]
+    fn test_parse_engine_string() {
+        
+        let my_os_str = OsStr::new("user:my:7b78a8e15d5ddfccaa71088ee44606981bb804d7");
+        let my_file_str = OsStr::new(r"file:C:\Microsoft.Autopilot.Security\MachineFunctionCerts\CY2TEAP00013459.CY2Test01.sst"); // have to prefix with "r" for raw string
+        let file_result = parse_engine_string(my_file_str).unwrap();
+        let os_result = parse_engine_string(my_os_str).unwrap();
+
+        // Verify file parse
+        assert!(matches!(file_result, OsProviderParameters::ContextFromFile{..}));
+        match file_result {
+            OsProviderParameters::ContextFromFile {file_path} => {
+                assert_eq!(file_path.as_os_str(), r"C:\Microsoft.Autopilot.Security\MachineFunctionCerts\CY2TEAP00013459.CY2Test01.sst");
+            }
+            _ => {}
+        }
+
+        // Verify store parse
+        assert!(matches!(os_result, OsProviderParameters::ContextFromStore{..}));
+
+        match os_result {
+            OsProviderParameters::ContextFromStore {store_name, is_machine, decoded_hex} => {
+                let expected_hex = hex::decode("7b78a8e15d5ddfccaa71088ee44606981bb804d7").unwrap();
+                assert_eq!(false, is_machine);
+                assert_eq!(store_name, "my".to_string());
+                assert_eq!(expected_hex, decoded_hex)
+            }
+            _ => {}
+        }
+    }
+
+    #[test]
+    fn test_os_provider_os_string() {
+        let my_os_str = OsStr::new("user:RustMy:3e2e13a694b3ed9e40849a4ab98b2c84d1b714d8");
```

Comment:
>3e2e13a694b3ed9e40849a4ab98b2c84d1b714d8 [](http://example.com/codeflow?start=48&length=40)

Another test using sha256.

Another expected failure where the hash length isn't 20 or 32.